crates/subspace-service/src/sync_from_dsn/import_blocks.rs

```diff
@@ -186,9 +186,10 @@ where
             while block_number.saturating_sub(best_block_number) >= QUEUED_BLOCKS_LIMIT.into() {
                 if !blocks_to_import.is_empty() {
                     // Import queue handles verification and importing it into the client
-                    import_queue_service
-                        .import_blocks(BlockOrigin::NetworkInitialSync, blocks_to_import.clone());
-                    blocks_to_import.clear();
+                    import_queue_service.import_blocks(
+                        BlockOrigin::NetworkInitialSync,
+                        std::mem::take(&mut blocks_to_import),
+                    );
```

Comment:
This makes performance worse than before and should be reverted.

`blocks_to_import` is allocated with correct capacity above. Since we need to give a vector of blocks to block import, we call `.clone()` that will only allocate as much memory as there are actual blocks in `blocks_to_import`, while `blocks_to_import` will remain allocated with correct capacity.

By replacing it with `mem::take()` you move original allocation into block import, while `blocks_to_import`'s capacity becomes zero, negating the capacity allocation that was set above.

We need to have two allocations one way or the other and the previous code was doing the minimum amount of work, while new code results in strictly more work and a lot more calls to the allocator.